src/main/java/com/xceptance/xlt/report/providers/CustomLogsReportProvider.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2005-2024 Xceptance Software Technologies GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.xceptance.xlt.report.providers;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileType;
+
+import com.xceptance.xlt.api.engine.Data;
+import com.xceptance.xlt.api.report.AbstractReportProvider;
+import com.xceptance.xlt.common.XltConstants;
+import com.xceptance.xlt.report.ReportGeneratorConfiguration;
+
+/**
+ * Report provider generating a report fragment about the custom data logs collected during the test run
+ */
+public class CustomLogsReportProvider extends AbstractReportProvider
+{
+
+    private static final String CUSTOM_DATA = "custom_data_logs";
+    
+    private Map<String, ZipOutputStream> foundScopes = new HashMap<String, ZipOutputStream>();
+    
+    private Path baseDir = null;
+    private Path targetDir = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object createReportFragment()
+    {
+        final CustomLogsReport report = new CustomLogsReport();
+
+        FileObject results = ((ReportGeneratorConfiguration) getConfiguration()).getResultsDirectory();
+        
+        baseDir = results.getPath();
```

Comment:
When creating the report from a result archive, this won't work as the results directory is inside the archive. 

To reproduce, simply create a zip file from your results directory and pass that to the report generator.